### PR TITLE
Support forwarding username to backends

### PR DESCRIPTION
**Pull request by zhaohuabing**
Related to https://github.com/envoyproxy/envoy/issues/32419

### CODE REVIEW

File: source/extensions/filters/http/basic_auth/basic_auth_filter.cc [hunk 1]
```diff
@@ -75,6 +77,10 @@ Http::FilterHeadersStatus BasicAuthFilter::decodeHeaders(Http::RequestHeaderMap&
                     "invalid_credential_for_basic_auth");
   }
 
+  if (!config_->forwardUsernameHeader().empty()) {
+    headers.addCopy(Http::LowerCaseString(config_->forwardUsernameHeader()), username);
```
[the remark this answers is not in the document]
Comment (htuch) [reply source/extensions/filters/http/basic_auth/basic_auth_filter.cc]: Can you add a test to capture this possible behavior as well? E.g. client has set, what happens now that we're doing the right thing?

**Comment by yanavlasov** [comment]
Pinging @htuch for final review.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**